mythril/mythril/mythril_config.py

```diff
@@ -0,0 +1,220 @@
+            os.mkdir(mythril_dir)
+
+        db_path = str(Path(mythril_dir) / "signatures.db")
+        if not os.path.exists(db_path):
+            # if the default mythril dir doesn't contain a signature DB
+            # initialize it with the default one from the project root
+            asset_dir = Path(__file__).parent.parent / "support" / "assets"
+            copyfile(str(asset_dir / "signatures.db"), db_path)
+
+        return mythril_dir
+
+    def _init_config(self) -> str:
+        """If no config file exists, create it and add default options.
+
+        Default LevelDB path is specified based on OS
+        dynamic loading is set to infura by default in the file
+        :return: LevelDB directory
+        """
+
+        leveldb_fallback_dir = self._get_fallback_dir()
+
+        if not os.path.exists(self.config_path):
+            log.info("No config file found. Creating default: " + self.config_path)
+            open(self.config_path, "a").close()
+
+        config = ConfigParser(allow_no_value=True)
+        # TODO: Remove this after this issue https://github.com/python/mypy/issues/2427 is closed
+        config.optionxform = str  # type:ignore
+        config.read(self.config_path, "utf-8")
+        if "defaults" not in config.sections():
+            self._add_default_options(config)
+
+        if not config.has_option("defaults", "leveldb_dir"):
+            self._add_leveldb_option(config, leveldb_fallback_dir)
+
+        if not config.has_option("defaults", "dynamic_loading"):
+            self._add_dynamic_loading_option(config)
+
+        with codecs.open(self.config_path, "w", "utf-8") as fp:
+            config.write(fp)
+
+        leveldb_dir = config.get(
+            "defaults", "leveldb_dir", fallback=leveldb_fallback_dir
+        )
+        return os.path.expanduser(leveldb_dir)
+
+    @staticmethod
+    def _get_fallback_dir() -> str:
+        """
+        Returns the LevelDB path
+        :return: The LevelDB path
+        """
+        system = platform.system().lower()
+        leveldb_fallback_dir = os.path.expanduser("~")
+        if system.startswith("darwin"):
+            leveldb_fallback_dir = os.path.join(
+                leveldb_fallback_dir, "Library", "Ethereum"
+            )
+        elif system.startswith("windows"):
+            leveldb_fallback_dir = os.path.join(
+                leveldb_fallback_dir, "AppData", "Roaming", "Ethereum"
+            )
+        else:
+            leveldb_fallback_dir = os.path.join(leveldb_fallback_dir, ".ethereum")
+        return os.path.join(leveldb_fallback_dir, "geth", "chaindata")
+
+    @staticmethod
+    def _add_default_options(config: ConfigParser) -> None:
+        """
+        Adds defaults option to config.ini
+        :param config: The config file object
+        :return: None
+        """
+        config.add_section("defaults")
+
+    @staticmethod
+    def _add_leveldb_option(config: ConfigParser, leveldb_fallback_dir: str) -> None:
```

Comment:
I think we should maybe separate the leveldb fallback dir functionality, looking at code using this function it is not immediately obvious how the parameter is used.
The only thing you see is
`self._add_leveldb_option(config, "somedir")

Reply:
You mean should we have a better function names and variables there? like we can change `leveldb_fallback_dir` to `leveldb_default_path` and function to `_add_leveldb_default_path()`

Reply:
I think `_add_level_db_default_path()` is a good suggestion